File: 228. Summary Ranges.py
class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        res = []
        if len(nums) > 0:
            begin, end = nums[0], nums[0]
            for i in xrange(1, len(nums)):
                if nums[i] == end + 1:
                    end = nums[i]
                else:
                    if begin != end:
                        res += [str(begin) + "->" +str(end)]
                    else:
                        res += [str(begin)]
                    begin, end = nums[i], nums[i]
            if begin != end:
                res += [str(begin) + "->" +str(end)]
            else:
                res += [str(begin)]
        return res

File: 228. Summary Ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        if (nums.size() == 0)
            return {};
        vector<string> res;
        int begin = nums[0], end = nums[0];
        for(int i = 1; i < nums.size(); i++)
        {
            if (nums[i] == end + 1)
                end = nums[i];
            else
            {
                if (begin != end)
                    res.push_back(to_string(begin) + "->" +to_string(end));
                else
                    res.push_back(to_string(begin));
                begin = nums[i];
                end = nums[i];
            }
        }
        if (begin != end)
            res.push_back(to_string(begin) + "->" +to_string(end));
        else
            res.push_back(to_string(begin));
        return res;
    }
};
